### PR TITLE
[IMP] edi_oca: Allow to use a sequence as part of exchange filename pattern

### PR DESCRIPTION
**Pull request by tuantrantg**

### CODE REVIEW

File: edi_oca/models/edi_exchange_type.py [hunk 1]
```diff
@@ -50,7 +50,9 @@ class EDIExchangeType(models.Model):
     direction = fields.Selection(
         selection=[("input", "Input"), ("output", "Output")], required=True
     )
-    exchange_filename_pattern = fields.Char(default="{record_name}-{type.code}-{dt}")
+    exchange_filename_pattern = fields.Char(
+        default="{record_name}-{type.code}-{dt}-{seq}"
```
Comment (simahawk) [inline edi_oca/models/edi_exchange_type.py]: IMO It shouldn't be here by default. However, we should have a help text to explain all avail variables.

File: edi_oca/models/edi_exchange_type.py [hunk 2]
```diff
@@ -130,6 +132,9 @@ class EDIExchangeType(models.Model):
             "Use it directly or within models rules (domain or snippet)."
         ),
     )
+    exchange_filename_sequence_id = fields.Many2one(
+        "ir.sequence", "Exchange Filename Sequence"
```
Comment (simahawk) [inline edi_oca/models/edi_exchange_type.py]: help txt pls

File: edi_oca/models/edi_exchange_type.py [hunk 3]
```diff
@@ -188,13 +193,22 @@ def _make_exchange_filename_datetime(self):
         now = datetime.now(utc).astimezone(tz)
         return slugify(now.strftime(date_pattern))
 
+    def _make_exchange_filename_sequence(self):
+        self.ensure_one()
+        return (
+            self.exchange_filename_sequence_id
+            and self.exchange_filename_sequence_id.next_by_id()
+            or ""
```
Comment (simahawk) [inline edi_oca/models/edi_exchange_type.py]:
```suggestion
            self.exchange_filename_sequence_id.next_by_id()
            if self.exchange_filename_sequence_id
            else ""
```

File: edi_oca/models/edi_exchange_type.py [hunk 4]
```diff
@@ -188,13 +193,22 @@ def _make_exchange_filename_datetime(self):
...
+        )
+
     def _make_exchange_filename(self, exchange_record):
         """Generate filename."""
         pattern = self.exchange_filename_pattern
         ext = self.exchange_file_ext
         if ext:
             pattern += ".{ext}"
         dt = self._make_exchange_filename_datetime()
+        seq = self._make_exchange_filename_sequence()
```
Comment (simahawk) [inline edi_oca/models/edi_exchange_type.py:227]: this requires a test

**Review by simahawk — CHANGES_REQUESTED** [review]
`[IMP] edi_oca: allow use of sequence for exchange filename pattern`

**Comment by tuantrantg** [comment]
hi @simahawk please review this PR again, thank you

**Review by simahawk — APPROVED** [review]

**Pull request merged**